### PR TITLE
Update demo

### PR DESCRIPTION
**Pull request by erwanMarmelab**
## Problem
React-Admin CRM demo was updated

## Solution
Update ra-supabase demo

### CODE REVIEW

File: packages/demo/src/misc/NbRelations.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { Typography, CircularProgress } from '@mui/material';
+import {
+    useRecordContext,
+    useReferenceManyFieldController,
+    useTimeout,
+} from 'ra-core';
```
Comment (slax57) [inline packages/demo/src/misc/NbRelations.tsx:6]:
```suggestion
} from 'react-admin';
```

File: packages/demo/src/notes/NewNote.tsx [hunk 2]
```diff
@@ -34,14 +34,17 @@ export const NewNote = ({
     if (!record || !identity) return null;
 
     const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+        console.log(record.type);
```
Comment (slax57) [inline packages/demo/src/notes/NewNote.tsx:37]: `console.log` spotted

File: packages/demo/src/notes/Note.tsx [hunk 3]
```diff
@@ -52,7 +51,6 @@ export const Note = ({
                 notify('Note deleted', { type: 'info', undoable: true });
                 update(reference, {
                     id: record.id,
-                    data: { nb_notes: record.nb_notes - 1 },
```
Comment (slax57) [inline packages/demo/src/notes/Note.tsx:55]: this `update` now does nothing, you should probably remove it altogether

File: packages/ra-supabase-ui-materialui/src/LoginPage.tsx [hunk 4]
```diff
@@ -72,7 +72,6 @@ import {
  * );
  */
 export const LoginPage = (props: LoginPageProps) => {
-    useRedirectIfAuthenticated();
```
Comment (slax57) [inline packages/ra-supabase-ui-materialui/src/LoginPage.tsx:75]: why did you remove this? :thinking:

**Review by slax57 — CHANGES_REQUESTED** [review]
Some questions already from reading the code. Now to test the demo.

But this is a huge PR anyways, incredible work! :muscle: :muscle:

File: packages/ra-supabase-ui-materialui/src/LoginPage.tsx (hunk 4 above)
Comment (adguernier) [reply packages/ra-supabase-ui-materialui/src/LoginPage.tsx:75]: Because this leads to an infinite redirection. Also, I think it doesn't make sense to redirect users if they want access this page: they might want to log in with other credentials.

Comment (slax57) [reply packages/ra-supabase-ui-materialui/src/LoginPage.tsx:75]: Kept, and added to the the changelog

**Comment by slax57** [comment]
All issues found in the demo should be fixed now! :white_check_mark:

**Review by slax57 — APPROVED** [review]

**Review by slax57 — APPROVED** [review]

**Review by djhi — APPROVED** [review]
:muscle:
